doi-requests-commons/src/main/java/no/unit/nva/doi/requests/service/impl/DynamoDBDoiRequestsService.java

```diff
@@ -97,23 +96,18 @@ public DynamoDBDoiRequestsService(AmazonDynamoDB client,
     @Override
     public List<DoiRequestSummary> findDoiRequestsByStatus(URI publisher, DoiRequestStatus status)
         throws ApiGatewayException {
-        RangeKeyCondition rangeKeyCondition = new RangeKeyCondition(DOI_REQUEST_STATUS_DATE);
-        rangeKeyCondition.beginsWith(status.toString());
-        ItemCollection<QueryOutcome> outcome;
-        try {
-            outcome = doiRequestsIndex.query(PUBLISHER_ID, publisher.toString(), rangeKeyCondition);
-        } catch (Exception e) {
-            throw new DynamoDBException(ERROR_READING_FROM_TABLE, e);
-        }
-        return parseJsonToDoiRequestSummaries(outcome);
+        List<Publication> publications = fetchPublicationsByPublisherAndStatus(publisher, status);
+        return publications
+            .stream().parallel()
+            .map(DoiRequestSummary::fromPublication).collect(Collectors.toList());
     }
 
     @Override
-    public List<DoiRequestSummary> findDoiRequestsByStatusAndOwner(URI publisher, DoiRequestStatus status, String owner)
+    public List<DoiRequestSummary> findDoiRequestsByStatusAndOwner(URI publisher, DoiRequestStatus status, String user)
```

Comment:
You can do later but ideally again a business method that belongs to `Publication` really.

Reply:
Not really. The misunderstanding comes from the fact that a DoiRequestSummary is basically a publication. So a Publication cannot return a list of Publications